### DISCUSSION TITLE
Multiple `find_with_related()`

### DISCUSSION THREAD

**Discussion by Kijewski** [SeaQL/sea-orm#1028] [Q&A] [answered]
Let's say I have a `parent::Entity` with zero or many `good_child::Entity`, `neutral_child::Entity`, and `chaotic_child::Entity`. (Just an example. The actual `child_*` tables have nothing in common with one another).

Each child has a field with its parent's id. Now I can select all good_children for some parent with

```rust
let (parent, good_children) = model::parent::Entity::find_by_id(42)
    .find_with_related(model::good_child::Entity)
    .all(&db.begin().await.map_err(…)?)
    .await
    .map_err(…)?
    .pop()
    .ok_or(…)?;
```

But how can I get the neutral and chaotic children in the same query? Simply adding another [`find_with_related()`] call does not work:

> no method named `find_with_related` found for struct `SelectTwoMany` in the current scope
> 
> method not found in `SelectTwoMany<…::parent::Entity, projektdb_model::good_child::Entity>`

---

I asked the same question on stackoverflow, too: [Multiple `find_with_related()` in sea_orm](https://stackoverflow.com/questions/73662449/multiple-find-with-related-in-sea-orm). I hope that's not too much noise.

[`find_with_related()`]: https://docs.rs/sea-orm/0.9.2/sea_orm/entity/prelude/struct.Select.html#method.find_with_related

**Comment by billy1624** [comment, marked as answer]
Hey @Kijewski, sorry for the delay. Currently we only support one `*_related` at a time. So, I'd suggest you to create a custom struct to hold the query result of each row consist of good_child, neutral_child and chaotic_child data. See https://www.sea-ql.org/SeaORM/docs/advanced-query/custom-select/#handling-custom-selects for details.

**Reply by Kijewski** [reply]
Thank you for your answer! I feared something like that was necessary. :) But well, it could be worse.

Thank you for the project! This was until now the only problem I had.

**Comment by ChoaibMouhrach** [comment]
@billy1624 is this feature added ?

**Reply by cnwzhu** [reply]
A typical query usually involves joining several tables, but the current functionality is difficult to use.

**Comment by JoelGarcia99** [comment]
Any update on this? Is this feature coming in the future?

**Reply by Reinhard-Berger** [reply]
Would be a Great feature

**Comment by cnwzhu** [comment]
Not only does find_also_related need to be called multiple times, but find_with_related, find_also_linked, and find_with_linked should also be able to be called multiple times. I don’t quite understand why the default design was set to relate to only one table in the query. If it can relate to one, why can’t it relate to multiple? It’s so strange. A typical query usually involves joining several tables, but the current functionality is difficult to use.

**Comment by nixpulvis** [comment]
Yea, this is a somewhat major deficiency. A very common use case for me is to have a relation setup like this: `A has-many B has-many C` and to want to load `A` with all of it's `B` and `C` at the same time. Breaking this into multiple queries is wasteful.
